.github/workflows/build-test-osx-m1-cirrus.yaml

```diff
@@ -0,0 +1,149 @@
+# This workflow builds and tests the semgrep-core binary for macOS M1
+# and generates the m1-wheel for pypi.
+
+# coupling: if you modify this file, modify also build-test-osx-x86.yaml
+# This file is mostly a copy-paste of build-test-osx-x86.yaml
+name: "[Test]: Build/Test OSX M1 - Cirrus Runners"
+
+on:
+  workflow_dispatch:
+    inputs:
+      use-cache:
+        description: "Use Opam Cache - uncheck the box to disable use of the opam cache, meaning a long-running but completely from-scratch build."
+        required: true
+        type: boolean
+        default: true
+  workflow_call:
+    inputs:
+      use-cache:
+        description: "Use Opam Cache - uncheck the box to disable use of the opam cache, meaning a long-running but completely from-scratch build."
+        required: false
+        type: boolean
+        default: true
+  pull_request: {}
+  schedule:
+    - cron: 27 * * * *
+
+jobs:
+  build-core-m1:
+    name: Build the OSX M1 binaries
+    runs-on:
+      [
+        "self-hosted",
+        "macOS",
+        "ARM64",
+        "ghcr.io/cirruslabs/macos-monterey-xcode:latest",
+      ]
+    env:
+      OPAM_SWITCH_NAME: "4.14.0"
+    steps:
+      - name: Setup runner directory
+        run: |
+          sudo mkdir /Users/runner
+          sudo chown admin:staff /Users/runner
+          sudo chmod 750 /Users/runner
+      - name: Brew Update & Install Python
+        run: |
+          brew update
+          brew install python@3.11
+      - uses: actions/checkout@v3
+        with:
+          submodules: true
+      # Note that this action handles both cache read and cache write.
+      # See https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows for more information on GHA caching.
+      - name: Cache Opam
+        id: cache-opam
+        uses: actions/cache@v3
+        if: ${{ inputs.use-cache || github.event_name == 'pull_request' || github.event_name == 'schedule' }}
+        env:
+          SEGMENT_DOWNLOAD_TIMEOUT_MINS: 2
+        with:
+          path: ~/.opam
+          key: "${{ runner.os }}-${{ runner.arch }}-${{ env.OPAM_SWITCH_NAME }}-opam-deps"
+      - name: Install dependencies
+        run: |
+          ./scripts/osx-setup-for-release.sh "${{ env.OPAM_SWITCH_NAME }}"
+      - name: Compile semgrep
+        run: |
+          opam exec -- make core
+          mkdir -p artifacts
+          cp ./bin/semgrep-core artifacts
+          zip -r artifacts.zip artifacts
+      - uses: actions/upload-artifact@v3
+        with:
+          path: artifacts.zip
+          name: semgrep-m1-${{ github.sha }}
+      - name: Save Semgrep Core Binary
+        id: cache-semgrep-core
+        uses: actions/cache/save@v3
+        with:
+          path: artifacts.zip
+          key: osx-m1-semgrep-core-VzlGZQINFX2
```

Comment:
```suggestion
          key: osx-m1-semgrep-core-{{github.sha}}
```

Reply:
We actually shouldn't be caching that all - removed.